### PR TITLE
Additional Reservoir Types Routing

### PR DESCRIPTION
**Pull request by jdmattern-noaa**
Hybrid and RFC Reservoir Types added with unit tests and integrated into routing. Routing tests included for Hybrid Persistence USGS type and RFC type reservoirs with the Croton NY domain.

## Additions
1. Updates to V02 routing and MC_Reach, nhd_io, along with Hybrid and RFC reservoir types.
2. Includes new Reservoir Parameter File along with USGS timeslice and RFC time series files.

## Testing

1. Reservoir Unit Tests and routing tests for Croton NY domain

## Notes

This PR replaces PR #302 because new commits did not automatically update in that PR.  

## Checklist

- [x] PR has an informative and human-readable title
- [x] Changes are limited to a single goal (no scope creep)
- [x] Code can be automatically merged (no conflicts)
- [x] Code follows project standards (link if applicable)
- [ ] Passes all existing automated tests
- [ ] Any _change_ in functionality is tested
- [ ] New functions are documented (with a description, list of inputs, and expected output)
- [ ] Placeholder code is flagged / future todos are captured in comments
- [ ] Visually tested in supported browsers and devices (see checklist below :point_down:)
- [ ] Project documentation has been updated (including the "Unreleased" section of the CHANGELOG)
- [ ] Reviewers requested with the [Reviewers tool](https://help.github.com/articles/requesting-a-pull-request-review/) :arrow_right:

### Target Environment support

- [ ] Windows
- [ ] Linux
- [ ] Browser

### Accessibility

- [ ] Keyboard friendly
- [ ] Screen reader friendly

### Other

- [ ] Is useable without CSS
- [ ] Is useable without JS
- [ ] Flexible from small to large screens
- [ ] No linting errors or warnings
- [ ] JavaScript tests are passing

### CODE REVIEW

**Comment by jdmattern-noaa** [comment]
@hellkite500 Is this todo still necessary? r already has the type of reservoir initialized and calls the particular type's (LP, hybrid, rfc) accordingly.
`
        #TODO: Add if isintance of the reservoir type
        #if isinstance(reservoir_object, lp_kernel):
`
https://github.com/NOAA-OWP/t-route/blob/158135eac49f4ba675c90a31d3af8949efd72b39/src/python_routing_v02/fast_reach/mc_reach.pyx#L897

**Comment by jameshalgren** [comment]
Thank you @jdmattern-noaa .

File: test/input/yaml/Croton_NY_rfc.yaml [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+        level_pool_weir_elevation: WeirE
+        level_pool_waterbody_max_elevation: LkMxE
+        level_pool_outfall_weir_coefficient: WeirC
+        level_pool_outfall_weir_length: WeirL
+        level_pool_overall_dam_length: DamL
+        level_pool_orifice_elevation: OrificeE
+        level_pool_orifice_coefficient: OrificeC
+        level_pool_orifice_area: OrificeA
+
+    hybrid_and_rfc:
```
Comment (jameshalgren) [inline test/input/yaml/Croton_NY_rfc.yaml:69]: @jdmattern-noaa @hellkite500 @mehdire61 
for the "hybrid" and "USGS" reservoirs, we need to build a list of the input files needed and summarize that in the documentation.

**Comment by jameshalgren** [comment]
Replaced by #324

**Pull request closed without merging**